src/pages/user/index.tsx

```diff
@@ -6,6 +6,38 @@ import RequestEmailChangeForm from 'components/users/request-email-change-form'
 import get from 'lodash/get'
 import SubscriptionDetails from 'components/users/subscription-details'
 
+const GithubConnectButton: React.FunctionComponent<{providers: string[]}> = ({
+  providers,
+}) => {
+  const {authToken} = useViewer()
+
+  const githubAlreadyConnected = providers.includes('github')
+
+  const buttonText = 'Connect your GitHub account'
+  const buttonStyles = 'text-white bg-blue-600 border-0 py-2 px-4 rounded'
+
+  if (githubAlreadyConnected) {
+    return (
+      <button
+        onClick={() => {}}
+        disabled
+        className={`${buttonStyles} opacity-50 cursor-not-allowed`}
+      >
+        {buttonText}
+      </button>
+    )
+  }
+
+  return (
+    <a
+      href={`${process.env.NEXT_PUBLIC_AUTH_DOMAIN}/users/github_passthrough?client_id=${process.env.NEXT_PUBLIC_CLIENT_ID}&auth_token=${authToken}`}
+      className={`${buttonStyles} focus:outline-none hover:bg-blue-700`}
+    >
+      {buttonText}
+    </a>
+  )
```

Comment:
This is what I came up with for functionally and visually displaying a "disabled" button (considering the active version is a link). It feels a little verbose, so I'm open to another approach. \cc @vojtaholik

Reply:
would it be possible to add this url to the `viewer` object (`minimal_user.jbuilder`) so we aren't doing complex url building client side?

What do you think about "already connected to github | remove connection" or similar versus a dead button. THis page desperately needs design support!

Reply:
Here is how it looks now when you've already connected:

<img width="957" alt="CleanShot 2021-02-08 at 10 15 26@2x" src="https://user-images.githubusercontent.com/694063/107248326-cea9b880-69f7-11eb-8362-b6517bf0891a.png">

I agree that we should explain the state of their GitHub connection if they have already connected. I have two variations in mind for that.

1. Leave the inactive button and put some text under it explaining:

<img width="955" alt="CleanShot 2021-02-08 at 10 22 11@2x" src="https://user-images.githubusercontent.com/694063/107248971-79ba7200-69f8-11eb-845a-f4a8fa4422aa.png">

2. Include only the text explanation and remove the button:

<img width="957" alt="CleanShot 2021-02-08 at 10 23 01@2x" src="https://user-images.githubusercontent.com/694063/107249005-82ab4380-69f8-11eb-8599-00baa08be638.png">

Reply:
> would it be possible to add this url to the `viewer` object (`minimal_user.jbuilder`) so we aren't doing complex url building client side?

I don't think it buys us much to shift the construction of this particular URL over to the server. The client_id and authToken are unique to egghead-next. It feels off to me to construct and serve a client-specific URL from the "general-purpose" API layer.